UnityProject/Assets/Scripts/Tilemaps/Matrix.cs

```diff
@@ -28,14 +29,44 @@ private void Start()
 
 		public bool IsPassableAt(Vector3Int origin, Vector3Int position)
 		{
-			return metaTileMap.IsPassableAt(origin, position);
+			if(origin.z != position.z)
+			{
+				//Uhhhhhh, error handling goes here?
+				return false;
+			}
+			//Check if it's a diagonal move
+			Vector3Int diff = position - origin;
+			int diffSum = diff.x + diff.y;
+			//Somewhat hacky way to get if it's diagonal, but it works
+			if(Math.Abs(diffSum) != 1)
+			{
+				//This is a DIAGONAL MOVEMENT!  Now we have to do four checks.
+				//This is confusing, basically there are two ways to travel diagonally so we have to check both of them
+				bool passable = true;
+				passable &= metaTileMap.IsPassableAt(origin, new Vector3Int(origin.x + diff.x, origin.y, origin.z));
```

Comment:
Is the use or origin.z a requirement from IsPassableAt? 
Because we tend to ignore Z most of the time...